internal/services/synapse/synapse_firewall_rule_resource.go

```diff
@@ -104,6 +104,30 @@ func resourceSynapseFirewallRuleCreateUpdate(d *pluginsdk.ResourceData, meta int
 		return fmt.Errorf("waiting on creation/update of %s: %+v", id, err)
 	}
 
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		return fmt.Errorf("context had no deadline")
+	}
+
+	stateChangeConf := &pluginsdk.StateChangeConf{
+		Pending: []string{string(synapse.ProvisioningStateProvisioning)},
+		Target:  []string{string(synapse.ProvisioningStateSucceeded)},
+		Refresh: func() (result interface{}, state string, err error) {
+			resp, err := client.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.Name)
+			if err != nil {
+				return nil, "Error", err
+			}
+			return resp, string(resp.ProvisioningState), err
+		},
+		MinTimeout:                30 * time.Second,
+		ContinuousTargetOccurence: 3,
+		Timeout:                   time.Until(deadline),
+	}
```

Comment:
is there a Swagger bug tracking this API issue? if the Future is being polled on (above) this should be `Succeeded` at the end of the LRO?

Reply:
Hi, I've added the swagger issue to the comment.

>  if the Future is being polled on (above) this should be Succeeded at the end of the LRO?
Yes, but it might not take effect immediately, so I add `ContinuousTargetOccurence: 3,` to ensure it's working.

Reply:
> Hi, I've added the swagger issue to the comment.

👍 

>> if the Future is being polled on (above) this should be Succeeded at the end of the LRO?
> Yes, but it might not take effect immediately, so I add ContinuousTargetOccurence: 3, to ensure it's working.

FWIW a Resource Manager operation should be entirely completed when the LRO is marked as `ProvisioningState: Succeeded`, the `ContinuousTargetOccurence` would be a workaround until the LRO is fixed.

Reply:
Yes, it's a workaround.